### PR TITLE
Week 4 branch

### PR DESCRIPTION
**Pull request by Maheima**

### CODE REVIEW

File: Week-4/assignment4_1.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const getNumber = () =>{
+    new Promise((res,rej) => {
```
Comment (jesusverma) [inline Week-4/assignment4_1.js]: YOU NEED TO RETURN THE PROMISE

File: Week-4/assignment4_1.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                rej(`Rejected with num ${randomNum}`)
+            }
+            res(`Resolved with num ${randomNum}`)
+            
+        }, randomNum * 10)
+    })
+}
+
+const x = getNumber()
+x.then()
```
Comment (jesusverma) [inline Week-4/assignment4_1.js]: X will now have pending prmosie , now next you have to either handle it or catch the error like...
x.then(()=>).catch(()=>)

File: Week-4/assignment4_2.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+Teacher.prototype.teach = function (subject, name,age) {
+    const t = Person.call(name, age);
+    
+    console.log(subject);
+
+}
+
+Object.setPrototypeOf(Teacher.prototype, Person.prototype)
```
Comment (jesusverma) [inline Week-4/assignment4_2.js]: through this you are overriding the teacher prototype , ideally this should come abpve line 12

File: Week-4/assignment4_2.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    this.name = name;
+    this.age = age;
+}
+
+
+function Teacher() {
+}
+
+Teacher.prototype.teach = function (subject, name,age) {
+    const t = Person.call(name, age);
```
Comment (jesusverma) [inline Week-4/assignment4_2.js]: there is no need to use call ,any reason why you are using call?

File: Week-4/assignment4_2.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    
+    console.log(subject);
```
Comment (jesusverma) [inline Week-4/assignment4_2.js]: you can able to access this.name and this.age here

Comment (Maheima) [reply Week-4/assignment4_2.js]: oh yes, it worked. Thanks Jesus. I don't know why the same code wasn't working in the morning

File: Week-4/assignment4_1.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const getNumber = () =>{
+    return new Promise((res,rej) => {
```
Comment (jesusverma) [inline Week-4/assignment4_1.js]: Please create your own custom promise and use here instead of JS promise.

**Pull request merged**